_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,378 @@
+import { ifError } from 'assert';
+import { mkdir, readFile, rmdir, writeFile } from 'fs/promises';
+import { EOL as newlineMarker } from 'os';
+import { sep as pathSep } from 'path';
+import { writeSync } from 'fs';
+import htmlStringify from 'rehype-stringify';
+import raw from 'rehype-raw';
+import remark2rehype from 'remark-rehype';
+import unified from 'unified';
+import visit from 'unist-util-visit';
+
+const require = createRequire(import.meta.url);
+const yaml = require('js-yaml');
+const markdown = require('remark-parse');
+const html = require('remark-html');
+const table2json = require('tabletojson').Tabletojson;
+const log = require('fancylog');
+const he = require('he');
+
+const DIR_DATA = './_data';
+const DIR_TEMP = './tmp';
+const REGEX_JS_INFO_STRS = /(javascript|js|mjs)/;
+const URL_RAW_PRPSLS_README =
+  'https://raw.githubusercontent.com/tc39/proposals/HEAD/README.md';
+
+// -----------------------------------------------------------------------------
+// Typedefs
+// -----------------------------------------------------------------------------
+
+/**
+ * @typedef {{
+ *   date: string,
+ *   url: string,
+ * }}
+ */
+const PresenceObj = {};
+
+// -----------------------------------------------------------------------------
+// Events
+// -----------------------------------------------------------------------------
+
+process.on('uncaughtException', (err /* , origin */) => {
+  const errorText = `Uncaught ${err instanceof Error ? err.name : 'error'}: ${
+    err.message
+  } (${err.code})`;
+  writeSync(
+    process.stderr.fd,
+    redden(`${UnicodeEscapes.errorSymbol}  ${errorText}${newlineMarker}`)
+  );
+});
+
+process.on('unhandledRejection', (wrn /* , origin */) => {
+  const warningText = `${wrn.name}: ${wrn.message}`;
+  writeSync(
+    process.stderr.fd,
+    yellow(`${UnicodeEscapes.warningSymbol}  ${warningText}${newlineMarker}`)
+  );
+});
+
+// -----------------------------------------------------------------------------
+// Helpers
+// -----------------------------------------------------------------------------
+
+/**
+ * Gets the URL value from the Markdown link key in the proposals README.
+ * @param {string} linkLabelLike
+ * @returns {string}
+ */
+function getUrlFromDoc(linkLabelLike) {
+  // log.debug('getUrlFromDoc linkLabelLike: ' + linkLabelLike);
+  const linkLabelLikeRegex = new RegExp(`\\[${linkLabelLike}.*\\]: (.*)`, 'g');
+  // log.debug({ linkLabelLikeRegex });
+  const results = linkLabelLikeRegex.exec(prpslsReadmeMarkdown);
+  // log.debug({ results });
+  return results[1];
+}
+
+/**
+ * Downloads the proposal's README file using the GitHub API, parses the
+ * Markdown file, and returns the contents of the JavaScript code block.
+ *
+ * If there are multiple matching JavaScript code blocks, the one that comes
+ * last in the document is used. If there are no matching JavaScript code
+ * blocks, an empty string will be used. (In such cases, a warning is emitted.)
```

Comment:
This has yet to be done &mdash; still thinking about how that warning to be emitted.

Choices:
- custom exception warning
- simply as a `log.warn`
- something else

Reply:
I don't think its so significant. The only thing is that this shouldn't overwrite a custom code block that has been added. In many cases, the existing code blocks have been added by hand and we may need to do that in the future still.

Reply:
Conclusion: perfectly acceptable to not warn if there are no code blocks.